src/node_buffer.cc

```diff
@@ -965,11 +965,9 @@ void IndexOfString(const FunctionCallbackInfo<Value>& args) {
   size_t result = haystack_length;
 
   if (enc == UCS2) {
-    String::Value needle_value(isolate, needle);
-    if (*needle_value == nullptr)
-      return args.GetReturnValue().Set(-1);
+    TwoByteValue needle_buffer(isolate, needle);
 
-    if (haystack_length < 2 || needle_value.length() < 1) {
+    if (haystack_length < 2 || needle_buffer.length()) {
```

Comment:
The new logic seems to be reversed comparing to what used to be before.
Should it rather say: `needle_buffer.length() == 0`?